pkg/cache/cluster.go

```diff
@@ -626,31 +773,50 @@ func (c *clusterCache) EnsureSynced() error {
 }
 
 // GetNamespaceTopLevelResources returns top level resources in the specified namespace
-func (c *clusterCache) GetNamespaceTopLevelResources(namespace string) map[kube.ResourceKey]*Resource {
-	c.lock.RLock()
-	defer c.lock.RUnlock()
-	resources := make(map[kube.ResourceKey]*Resource)
-	for _, res := range c.nsIndex[namespace] {
-		if len(res.OwnerRefs) == 0 {
-			resources[res.ResourceKey()] = res
+func (c *clusterCache) GetNamespaceTopLevelResources(namespace string) *Resources {
+	res, _ := c.LoadNsIndex(namespace)
```

Comment:
No error check here too. That's why I'm suggesting to just let it panic if the field contents is corrupted. The next line will panic anyway.